src/features/favorites/pages/Filters.tsx

```diff
@@ -0,0 +1,160 @@
+import { t } from '@lingui/macro'
+import { useNavigation } from '@react-navigation/native'
+import debounce from 'lodash.debounce'
+import React, { useRef } from 'react'
+import { Linking, ScrollView, ViewStyle } from 'react-native'
+import styled from 'styled-components/native'
+
+import { ReinitializeFilters } from 'features/favorites/atoms/Buttons/ReinitializeFilters'
+import { useFavoritesState } from 'features/favorites/pages/FavoritesWrapper'
+import { GeolocPermissionState, useGeolocation } from 'libs/geolocation'
+import { GeolocationActivationModal } from 'libs/geolocation/components/GeolocationActivationModal'
+import { IGeolocationContext } from 'libs/geolocation/GeolocationWrapper'
+import { _ } from 'libs/i18n'
+import { ButtonPrimary } from 'ui/components/buttons/ButtonPrimary'
+import { PageHeader } from 'ui/components/headers/PageHeader'
+import { useModal } from 'ui/components/modals/useModal'
+import { Validate } from 'ui/svg/icons/Validate'
+import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
+import { ACTIVE_OPACITY } from 'ui/theme/colors'
+
+export const DEBOUNCED_CALLBACK = 200
+
+interface Options extends IGeolocationContext {
+  showGeolocPermissionModal: () => void
+}
+
+export const useSelectFilter = ({
+  showGeolocPermissionModal,
+  position,
+  permissionState,
+  requestGeolocPermission,
+}: Options) => {
+  const { filter: stateFilter, dispatch } = useFavoritesState()
+
+  function selectFilter(filter: keyof typeof FILTER_OPTIONS | undefined) {
+    return () => dispatch({ type: 'SET_FILTER', payload: filter })
```

Comment:
N'est-ce pas plus clair de cette manière ?
```suggestion
    return () => dispatch({ type: 'SET_FILTER', filter })
```

Reply:
agree

Reply:
c'était pour avoir un type consistant à travers toutes les actions. Communément, on a type et payload.

Reply:
Même remarque, code convention, cela peut être corrigé dans un futur commit de refactor pour éviter les inconsistances.